sources/api/sundog/src/main.rs

```diff
@@ -221,6 +224,54 @@ where
     Ok(populated_settings)
 }
 
+// Builds the proxy environment variables to pass to settings generators
+async fn build_proxy_env<P>(socket_path: P) -> Result<HashMap<String, String>>
+where
+    P: AsRef<Path>,
+{
+    // Retrieve network proxy-related settings.
+    let settings: serde_json::Value =
+        schnauzer::get_json(&socket_path, "/settings", Some(("prefix", "network")))
+            .await
+            .context(error::RetrieveSettingsSnafu)?;
+    let mut proxy_envs = HashMap::new();
+    if let Some(network_settings) = settings.get("network") {
+        if let Some(https_proxy) = network_settings.get("https-proxy").and_then(|s| s.as_str()) {
+            proxy_envs.insert("https_proxy".to_string(), https_proxy.to_string());
+            proxy_envs.insert("HTTPS_PROXY".to_string(), https_proxy.to_string());
+        }
+        if let Some(no_proxy) = network_settings.get("no-proxy").and_then(|v| v.as_array()) {
+            let mut no_proxy: Vec<String> = no_proxy
+                .iter()
+                .map(|s| s.as_str().unwrap_or_default().to_string())
+                .filter(|s| !s.is_empty())
+                .collect();
+            no_proxy.push("localhost".to_string());
+            no_proxy.push("127.0.0.1".to_string());
+            // We potentially need to also no-proxy some K8s related domains for K8s variants
+            let settings: serde_json::Value =
+                schnauzer::get_json(socket_path, "/settings", Some(("prefix", "kubernetes")))
+                    .await
+                    .context(error::RetrieveSettingsSnafu)?;
```

Comment:
I think we need to `NO_PROXY` these even if we don't have an explicit `settings.network.no-proxy` setting. Do we also need to remove duplicate values from the array, in case these were already set?

(I find an aggressive approach to outdenting and early returns helps to catch logic bugs like this, if it is a bug, or to focus the comment on why the logic actually depends on the parent condition if it's not obvious.)

Reply:
Nice catch!

> Do we also need to remove duplicate values from the array, in case these were already set?

I believe duplicate entries are OK from what we've tested before with `host-ctr` (aws go sdk).We also don't prevent duplicate values in the rendered `proxy.env` systemd `EnvironmentFile`.